### PR TITLE
Update CC=gcc setting for Fedora s390x

### PR DESCRIPTION
**Pull request by Jakob-Naucke**
For FFI to C in Go, `CC=gcc` is required on Fedora s390x (see https://github.com/kata-containers/osbuilder/issues/217).

- osbuilder: Remove CC=gcc for Fedora s390x (no longer required for Rust agent)
- runtime: Use CC=gcc on Fedora s390x (is still required and was never added for runtime)

Fixes: #1973

### CODE REVIEW

File: src/runtime/arch/s390x-options.mk [hunk 1]
```diff
@@ -11,3 +11,10 @@ MACHINEACCELERATORS :=
 CPUFEATURES :=
 
 QEMUCMD := qemu-system-s390x
+
+# See https://github.com/kata-containers/osbuilder/issues/217
+FEDORA_LIKE = $(shell grep fedora /etc/os-release 2> /dev/null)
```
Comment (jodh-intel) [inline src/runtime/arch/s390x-options.mk:16]: Might be overkill but you could anchor this with `... egrep "\<fedora\>" ...`.

**Review by jodh-intel — APPROVED** [review]
Thanks @Jakob-Naucke.

File: src/runtime/arch/s390x-options.mk (hunk 1 above)
Comment (fidencio) [reply src/runtime/arch/s390x-options.mk:16]: This is going to match CentOS, AlmaLinux, Rocky Linux, RHEL, OEL, etc.
I guess this is **exactly** what you're trying to do, but I'd like a confirmation from you, @Jakob-Naucke.

Comment (Jakob-Naucke) [reply src/runtime/arch/s390x-options.mk:16]: > Might be overkill but you could anchor this with `... egrep "\<fedora\>" ...`.

Updated.

> This is going to match CentOS, AlmaLinux, Rocky Linux, RHEL, OEL, etc.
> I guess this is *exactly* what you're trying to do, but I'd like a confirmation from you, @Jakob-Naucke.

Most of those don't package for s390x :stuck_out_tongue_winking_eye: but I also just tried this on ClefOS (~ s390x's CentOS) and it needs it too. So yes.

**Review by fidencio — APPROVED** [review]
lgtm, thanks @Jakob-Naucke!

**Pull request merged**